src/runtime/internal/style_manager.ts

```diff
@@ -71,10 +72,8 @@ export function clear_rules() {
 	raf(() => {
 		if (active) return;
 		managed_styles.forEach(info => {
-			const { stylesheet } = info;
-			let i = stylesheet.cssRules.length;
-			while (i--) stylesheet.deleteRule(i);
-			info.rules = {};
+			const { style_element } = info;
+			detach(style_element);
 		});
 		managed_styles.clear();
 	});
```

Comment:
Just thinking out loud here, but could this `raf(() => {})` callback be the reason as to why the extra `raf.tick()` is needed? @tanhauhau

Reply:
yes. this is the reason why. the cleanup happens on the next raf callback after the transition is done.